File: 824.py
"""
Note the vowel can be lowercase or uppercase
"""
class Solution(object):
    def toGoatLatin(self, S):
        """
        :type S: str
        :rtype: str
        """
        res = []
        vowels = ['a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U']
        words = S.split(' ')
        for i in range(len(words)):
            word = words[i]
            if word[0] in vowels:
                newWord = word + 'ma' + 'a' * (i+1)
            else:
                newWord = word[1:] + word[0] + 'ma' + 'a' * (i+1)
            res.append(newWord)
        return " ".join(res)

File: solution/go/824.go
/*
Use a map to quickly look up the vowels.
*/
func toGoatLatin(S string) string {
    vowels := map[string]int{"a": 1, "e": 1, "i": 1, "o": 1, "u": 1,
                             "A": 1, "E": 1, "I": 1, "O": 1, "U": 1}

    words := strings.Split(S, " ")
    for i, word := range words {
        if _, ok := vowels[string(word[0])]; !ok {
            words[i] = word[1:] + string(word[0])
        }
        words[i] += "ma" + strings.Repeat("a", i + 1)
    }

    return strings.Join(words, " ")
}
